raiden_libs/utils/signing.py

```diff
@@ -56,63 +63,87 @@ def format_int(value, size):
     return msg
 
 
-def keccak256(*args) -> bytes:
-    return keccak(pack(*args))
-
-
-def sign(privkey: str, msg: bytes, v=0) -> bytes:
-    assert isinstance(msg, bytes)
-    assert isinstance(privkey, str)
-
-    pk = PrivateKey.from_hex(remove_0x_prefix(privkey))
-    assert len(msg) == 32
-
-    sig = pk.sign_recoverable(msg, hasher=None)
-    assert len(sig) == 65
-
-    sig = sig[:-1] + bytes([sig[-1] + v])
-
-    return sig
-
-
-def sign_data(privkey: str, msg: bytes, v=27):
-    pk = PrivateKey.from_hex(remove_0x_prefix(privkey))
-    sha3 = lambda x: keccak_256(x).digest()
-    sig = pk.sign_recoverable(msg, hasher=sha3)
-    return sig[:-1] + chr(sig[-1] + v).encode()
+def keccak256(*args, hasher: Hasher=sha3) -> bytes:
+    """
+    Simulates Solidity's keccak256 packing. Integers can be passed as tuples where the second tuple
+    element specifies the variable's size in bits, e.g.:
+    keccak256((5, 32))
+    would be equivalent to Solidity's
+    keccak256(uint32(5))
+    Default size is 256.
+    """
+    if hasher:
+        return hasher(pack(*args))
+    else:
+        return pack(*args)
 
 
 def public_key_to_address(public_key: Union[PublicKey, bytes]) -> Address:
     """ Converts a public key to an Ethereum address. """
     if isinstance(public_key, PublicKey):
         public_key = public_key.format(compressed=False)
     assert isinstance(public_key, bytes)
-    return encode_hex(keccak(public_key[1:])[-20:])
+    return to_checksum_address(sha3(public_key[1:])[-20:])
 
 
-def private_key_to_address(private_key: str) -> Address:
+def private_key_to_address(private_key: Union[str, bytes]) -> Address:
     """ Converts a private key to an Ethereum address. """
-    return to_checksum_address(
-        public_key_to_address(PrivateKey.from_hex(remove_0x_prefix(private_key)).public_key),
-    )
+    if isinstance(private_key, str):
+        private_key = to_bytes(hexstr=private_key)
+    pk = PrivateKey(private_key)
+    return public_key_to_address(pk.public_key)
 
 
-def address_from_signature(sig: bytes, msg: bytes) -> Address:
+def address_from_signature(data: bytes, signature: bytes, hasher: Hasher=sha3) -> Address:
     """Convert an EC signature into an ethereum address"""
-    assert len(sig) == 65
+    if not isinstance(signature, bytes) or len(signature) != 65:
+        raise InvalidSignature('Invalid signature, must be 65 bytes')
     # Support Ethereum's EC v value of 27 and EIP 155 values of > 35.
-    if sig[-1] >= 35:
-        network_id = (sig[-1] - 35) // 2
-        sig = sig[:-1] + bytes([sig[-1] - 35 - 2 * network_id])
-    elif sig[-1] >= 27:
-        sig = sig[:-1] + bytes([sig[-1] - 27])
+    if signature[-1] >= 35:
```

Comment:
Can you add these numbers, `35`, `65` e.t.c. as constants somewhere and resue them from there?

Reply:
These values are hardcoded by spec, so I don't think they're worth to be put somewhere else, and `raiden-libs` doesn't have a `constants.py`, we would need to create one only for it. We can discuss it after, and make it in another PR, if needed.